languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -39150,6 +39150,100 @@ though, and has since been slightly modified)-->
+        <rulegroup id="VIRGULE_DEBUT_DE_PHRASE_3" name="virgule debut de phrase" default="temp_off">
+            <rule>
+                <pattern>
+                    <token postag="SENT_START"/>
+                    <marker>
+                        <token regexp="yes">bon(jour|soir)|salut|enchantée?s?|bienvenue?s?
+                            <exception postag="P.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="N.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="D.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="Z.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="C coor.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="R pers obj.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="J.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="A.*" postag_regexp="yes" scope="next"/>
+                        </token>
+                    </marker>
+                    <token regexp="yes">[^,‘’–)\[—.!"\/;\]\|:(\…'«»“”-]+</token>
+                </pattern>
+                <message>Une virgule semble être requise dans ce contexte.</message>
+                <suggestion>\2,</suggestion>
+                <example correction="Bonjour,"><marker>Bonjour</marker> je m'appelle Udo.</example>
+                <example>Salut!</example>
+                <example>Bonjour chez vous. (P)</example>
+                <example>Bonjour camarade. (N)</example>
+                <example>Bonjour les amis. (D)</example>
+                <example>Bonjour Jean. (Z)</example>
+                <example>Bonjour et bienvenue. (C coor)</example>
+                <example>Bonjour toi. (R pers obj)</example>
+                <example>Bonjour chers amis. (J)</example>
+                <example>Bonjour toutes et tous. (A)</example>
+            </rule>
+            <rule>
+                <pattern>
+                    <token postag="SENT_START"/>
+                        <token regexp="yes">bonnes?</token>
+                    <marker>
+                        <token regexp="yes">nuits?|journées?|vacances|après-midi|matinées?
+                            <exception postag="P.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="N.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="D.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="Z.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="C coor.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="R pers obj.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="J.*" postag_regexp="yes" scope="next"/>
+                            <exception postag="A.*" postag_regexp="yes" scope="next"/>
```

Comment:
You can tidy this up:

`<exception postag="[PNDZJA].*|C coor.*|R pers obj.*" postag_regexp="yes" scope="next"/>`.

Or you can go right ahead and say what you want the next token to be (`R pers suj`?). Do you really want to exclude all the others?

"Bonjour mon ami !" seems to be missing a comma, as does "Bonne nuit camarade !"